### DISCUSSION TITLE
Slow selection operation in xarray

### DISCUSSION THREAD

**Discussion by mlambar** [pydata/xarray#6291] [Q&A] [answered]
Hello,
I am trying to select specific values from a dataset in xarray and append it in a dataframe. However , it takes almost 20 minutes for this operation. Is there a way to optimize this?

Here's a snippet of code of what I'm doing : 
```
import xarray as xr
import pandas as pd 

df1 = xr.open_mfdataset(r'...\E*.nc', parallel = True)  
df2= pd.DataFrame(columns={'Date','lat' , 'lon' , 'E'} ).set_index('Date')
for i,j in zip(df3.Latitude, df3.Longitude): 
      df2 = df2.append(df1['E'].sel(lat= i, lon= j, method='nearest').to_dataframe())

```
Thank you!

**Comment by dcherian** [comment, marked as answer]
I  recommend  reading https://docs.xarray.dev/en/stable/user-guide/indexing.html#more-advanced-indexing, subsetting all values you want out of `df1` and then calling  `to_dataframe` on the result.
